src/main/java/eu/webeid/security/validator/AuthTokenValidatorBuilder.java

```diff
@@ -163,7 +179,10 @@ public AuthTokenValidatorBuilder withDesignatedOcspServiceConfiguration(Designat
      */
     public AuthTokenValidator build() throws NullPointerException, IllegalArgumentException, JceException {
         configuration.validate();
-        return new AuthTokenValidatorImpl(configuration);
+        if (ocspClient == null) {
+            ocspClient = OcspClientImpl.build(configuration.getOcspRequestTimeout());
```

Comment:
Maybe it makes sense to rename OcspClientImpl to OkHttpOcspClient?

Reply:
> Maybe it makes sense to rename OcspClientImpl to OkHttpOcspClient?

Thanks, good point! Done.

> configuration.getOcspRequestTimeout() probably could be removed because those needing it can provide a preconfigured OkHttpOcspClient?

We have to retain API stability and I would imagine there are folks who don't need much other control besides the timeout.

> Maybe OkHttpOcspClient should also give a possibility to configure the OkHttp inside of it, so that request timeouts, proxies, etc could be enabled

I think advanced configuration needs are better served by providing a preconfigured custom instance that derives from OkHttpOcspClient via withOcspClient().

Do you agree or am I missing something?

Reply:
Agree with the last point, but from the last commit I don't see how somebody would provide a preconfigured OkHttpClient to OkHttpOcspClient? Probably the constructor should be made public?

Reply:
Thanks for noticing, done! The whole `OkHttpOcspClient` was initially intended to be an implementation detail, but I guess there is no better way than to make it open for extension considering the advanced configuration use case.